src/EPR.Calculator.Frontend/Controllers/DefaultParametersController.cs

```diff
@@ -8,58 +8,66 @@ namespace EPR.Calculator.Frontend.Controllers
-                    ViewBag.SchemeSetupCosts = CalculateTotal(defaultSchemeParameters, ParameterType.SchemeSetupCosts, true);
-                    ViewBag.LateReportingTonnage = CalculateTotal(defaultSchemeParameters, ParameterType.LateReportingTonnage);
-                    ViewBag.MaterialityThreshold = CalculateTotal(defaultSchemeParameters, ParameterType.MaterialityThreshold);
-                    ViewBag.BadDebtProvision = CalculateTotal(defaultSchemeParameters, ParameterType.BadDebtProvision);
-                    ViewBag.Levy = CalculateTotal(defaultSchemeParameters, ParameterType.Levy);
-                    ViewBag.TonnageChange = CalculateTotal(defaultSchemeParameters, ParameterType.TonnageChangeThreshold);
-                    ViewBag.EffectiveFrom = defaultSchemeParameters[0].EffectiveFrom;
+                    if (defaultSchemeParameters != null)
+                    {
+                        this.ViewBag.CommunicationData = this.CalculateTotal(defaultSchemeParameters, ParameterType.CommunicationCosts, true);
+                        this.ViewBag.OperatingCosts = this.CalculateTotal(defaultSchemeParameters, ParameterType.SchemeAdministratorOperatingCosts, true);
+                        this.ViewBag.PreparationCosts = this.CalculateTotal(defaultSchemeParameters, ParameterType.LocalAuthorityDataPreparationCosts, true);
+                        this.ViewBag.SchemeSetupCosts = this.CalculateTotal(defaultSchemeParameters, ParameterType.SchemeSetupCosts, true);
+                        this.ViewBag.LateReportingTonnage = this.CalculateTotal(defaultSchemeParameters, ParameterType.LateReportingTonnage);
+                        this.ViewBag.MaterialityThreshold = this.CalculateTotal(defaultSchemeParameters, ParameterType.MaterialityThreshold);
+                        this.ViewBag.BadDebtProvision = this.CalculateTotal(defaultSchemeParameters, ParameterType.BadDebtProvision);
+                        this.ViewBag.Levy = this.CalculateTotal(defaultSchemeParameters, ParameterType.Levy);
+                        this.ViewBag.TonnageChange = this.CalculateTotal(defaultSchemeParameters, ParameterType.TonnageChangeThreshold);
+                        this.ViewBag.EffectiveFrom = defaultSchemeParameters[0].EffectiveFrom;
 
-                    ViewBag.IsDataAvailable = true;
+                        this.ViewBag.IsDataAvailable = true;
 
-                    return View();
+                        return this.View();
+                    }
                 }
 
                 if (response.StatusCode == HttpStatusCode.NotFound)
                 {
-                    ViewBag.IsDataAvailable = false;
-                    return View();
+                    this.ViewBag.IsDataAvailable = false;
+                    return this.View();
                 }
 
-                return RedirectToAction(ActionNames.StandardErrorIndex, "StandardError");
+                return this.RedirectToAction(ActionNames.StandardErrorIndex, "StandardError");
```

Comment:
Can we create a readonly constant for Actions and add

- "ParameterUploadFile"
-  "StandardError"

Reply:
For Controllers?

Reply:
Sorry about this. No action needed